src/KingSystem/Map/mapObject.cpp

```diff
@@ -1,7 +1,373 @@
+    if (!mLinkData)
+        return nullptr;
+
+    ObjectLink* result = mLinkData->mLinksToSelf.findLinkWithType(MapLinkDefType::PlacementLOD);
+    if (result)
+        return result->other_obj;
+
+    return nullptr;
+}
+
+Object* Object::findPlacementLODLinkObject() const {
+    if (!mLinkData)
+        return nullptr;
+
+    ObjectLink* result = mLinkData->mLinksToSelf.findLinkWithType(MapLinkDefType::PlacementLOD);
+    if (result)
+        return result->other_obj;
+
+    return nullptr;
+}
+
+const char* Object::getHashIdStringDebug() const {
+    act::ActorDebug::HashUnused t;
+
+    const char* terminator;
+
+    if (mMubinIter.tryGetParamUIntByKey(&t.hash, "HashId")) {
+        terminator = act::ActorDebug::instance()->getNullStr(&t);
+    } else
+        terminator = "";
+
+    return terminator;
+}
+
+const char* Object::getHashIdStringDebug_0() const {
+    act::ActorDebug::HashUnused t;
+
+    const char* terminator;
+
+    if (mMubinIter.tryGetParamUIntByKey(&t.hash, "HashId")) {
+        terminator = act::ActorDebug::instance()->getNullStr(&t);
+    } else
+        terminator = "";
+
+    return terminator;
+}
+
+void* Object::getRails() const {
+    if (!mLinkData)
+        return nullptr;
+
+    return mLinkData->mRails;
+}
+
+void* Object::getRails_0() const {
+    if (!mLinkData)
+        return nullptr;
+
+    return mLinkData->mRails;
+}
+
+bool Object::allocLinkData(sead::ExpHeap* heap) {
+    mLinkData = new (heap) ObjectLinkData;
+
+    return mLinkData != nullptr;
+}
+
+bool Object::hasGenGroup() const {
+    if (!mLinkData)
+        return false;
+
+    return mLinkData->mGenGroup != nullptr;
+}
+
+void Object::incrementLinkNum() {
+    if (mNumLinksPointingToMe != 0xFF) {
+        mNumLinksPointingToMe += 1;
+        return;
+    }
+
+    auto name = sead::SafeString(getUnitConfigName());
+
+    if (name == "LinkTagNone") {
+        // More than 255 actors are specified in one generation group.
+        // If it is not a test, consult the programmer.
+        auto msg = sead::SafeString("一つの生成グループに 255 以上のアクタが指定されています。"
+                                    "テストで無ければプログラマに相談");
+        printDebugMsg(this, &msg, nullptr);
+        return;
+    }
+
+    // We have exceeded the expected number of links.
+    // If it's not a test, talk to your programmer.
+    auto msg = sead::SafeString("リンクの想定数を越えました。テストで無ければプログラマに相談");
+    printDebugMsg(this, &msg, nullptr);
+}
+
+void Object::decrementLinkNum() {
+    if (mNumLinksPointingToMe != 0)
+        --mNumLinksPointingToMe;
+}
+
+sead::Vector3f Object::getRotate() const {
+    sead::Vector3f result;
+    mMubinIter.getRotate(&result);
+
+    return result;
+}
+
+// NON_MATCHING: Vec3f copy incorrect
+void Object::getTraversePosAndLoadDistance(sead::Vector3f* traversePos, f32* loadDist) const {
+    if (mFlags.isOn(Flag::HasTraversePos)) {
+        mMubinIter.tryGetFloatArrayByKey(&traversePos->x, "TraversePos");
+    } else {
+        *traversePos = mTranslate;
```

Comment:
does `traversePos->x = mTranslate.x; /* same for y, z */` fix it?

Reply:
no dice :(